### PR TITLE
[release/9.0] Emscripten version reference for RTM build

### PR DESCRIPTION
**Pull request by pavel-purma**
Added reference to Microsoft.NET.Workload.Emscripten.Current.Manifest and use that version fro Emscripten packages for RTM build

### CODE REVIEW

**Review by mmitche — APPROVED** [review]

File: eng/Versions.props [hunk 1]
```diff
@@ -240,7 +240,8 @@
                like - DarcDependenciesChanged.Microsoft_NET_Workload_Emscripten_Current_Manifest-9_0_100_Transport
     -->
     <MicrosoftNETWorkloadEmscriptenCurrentManifest90100TransportVersion>9.0.0-rc.2.24421.3</MicrosoftNETWorkloadEmscriptenCurrentManifest90100TransportVersion>
-    <MicrosoftNETRuntimeEmscriptenVersion>$(MicrosoftNETWorkloadEmscriptenCurrentManifest90100TransportVersion)</MicrosoftNETRuntimeEmscriptenVersion>
+    <MicrosoftNETWorkloadEmscriptenCurrentManifest90100Version>9.0.0-rc.2.24421.3</MicrosoftNETWorkloadEmscriptenCurrentManifest90100Version>
+    <MicrosoftNETRuntimeEmscriptenVersion>$(MicrosoftNETWorkloadEmscriptenCurrentManifest90100Version)</MicrosoftNETRuntimeEmscriptenVersion>
```
Comment (mmitche) [inline eng/Versions.props:244]: This is because there are references to the packages via stable versions?

Comment (pavel-purma) [reply eng/Versions.props:244]: TransportVersion is coming from non-shipping dependency and that did not work when used for references with version defined by MicrosoftNETRuntimeEmscriptenVersion.  I have added another dependency with shipping versioning (MicrosoftNETWorkloadEmscriptenCurrentManifest90100Version) and use that version as for further references of Emscripten.

Comment (lewing) [reply eng/Versions.props]: Was the source build version not sufficient?

**Comment by carlossanlop** [comment]
Today's the RC2 snap. There's a merge conflict and an open conversation. @pavel-purma can you please unblock the PR so we can get it merged?

**Review by lewing — CHANGES_REQUESTED** [review]
This should no longer be needed (for rtm at least).  See the discussion here https://github.com/dotnet/sdk/pull/43554

The manifests change name when they go to rtm now, not when the versions and as a result the same name is calculated from the stable version and transport package version in rtm.

**Review by carlossanlop — CHANGES_REQUESTED** [review]
@pavel-purma - Please retarget this PR to the [release/9.0-staging](https://github.com/dotnet/runtime/tree/release/9.0-staging) branch.

You can retarget to that branch by clicking on the Edit button on the top right (next to the PR title) and choosing the release/9.0-staging branch from the dropdown. Important: Please make sure you don't bring any unrelated changes from the release/9.0 branch when retargeting to release/9.0-staging.

**Pull request closed without merging**